tool/tctl/common/bots_command.go

```diff
@@ -546,6 +526,296 @@ func (c *BotsCommand) UpdateBot(ctx context.Context, client *authclient.Client)
+			return trace.BadParameter("token %q is valid for bot with name %q, not %q",
+				c.tokenID, token.GetBotName(), c.botName)
+		}
+	}
+
+	return trace.Wrap(outputToken(c.stdout, c.format, client, bot, token))
+}
+
+var showMessageTemplate = template.Must(template.New("show").Funcs(template.FuncMap{
+	"bold": bold,
+}).Parse(`Bot: {{.instance.Spec.BotName}}
+ID:  {{.instance.Spec.InstanceId}}
+
+Initial Authentication: {{.initial_authentication_table}}
+
+Latest Authentication: {{.latest_authentication_table}}
+
+Latest Heartbeat: {{.heartbeat_table}}
+
+To view a full, machine-readable record including past heartbeats and
+authentication records, run:
+
+> {{.executable}} get bot_instance/{{.instance.Spec.BotName}}/{{.instance.Spec.InstanceId}}
+
+To onboard a new instance for this bot, run:
+
+> {{.executable}} bots instances add {{.instance.Spec.BotName}}
+`))
+
+func (c *BotsCommand) ShowBotInstance(ctx context.Context, client *authclient.Client) error {
+	botName, instanceID, err := parseInstanceID(c.instanceID)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	instance, err := client.BotInstanceServiceClient().GetBotInstance(ctx, &machineidv1pb.GetBotInstanceRequest{
+		BotName:    botName,
+		InstanceId: instanceID,
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	initialAuthenticationTable := formatBotInstanceAuthentication(instance.Status.InitialAuthentication)
+
+	var latestAuthenticationTable string
+	if len(instance.Status.LatestAuthentications) > 0 {
+		latest := instance.Status.LatestAuthentications[len(instance.Status.LatestAuthentications)-1]
+		latestAuthenticationTable = formatBotInstanceAuthentication(latest)
+	} else {
+		latestAuthenticationTable = "No authentication records."
+	}
+
+	var heartbeatTable string
+	if len(instance.Status.LatestHeartbeats) > 0 {
+		latest := instance.Status.LatestHeartbeats[len(instance.Status.LatestHeartbeats)-1]
+		heartbeatTable = formatBotInstanceHeartbeat(latest)
+	} else {
+		heartbeatTable = "No heartbeat records."
+	}
+
+	templateData := map[string]interface{}{
+		"executable":                   os.Args[0],
+		"instance":                     instance,
+		"initial_authentication_table": initialAuthenticationTable,
+		"latest_authentication_table":  latestAuthenticationTable,
+		"heartbeat_table":              heartbeatTable,
+	}
+
+	return trace.Wrap(showMessageTemplate.Execute(os.Stdout, templateData))
+}
+
+// botJSONResponse is a response generated by the `tctl bots add` family of
+// commands when the format is `json`
+type botJSONResponse struct {
+	UserName string        `json:"user_name"`
+	RoleName string        `json:"role_name"`
+	TokenID  string        `json:"token_id"`
+	TokenTTL time.Duration `json:"token_ttl"`
+}
+
+// outputToken writes token information to stdout, depending on the token format.
+func outputToken(wr io.Writer, format string, client *authclient.Client, bot *machineidv1pb.Bot, token types.ProvisionToken) error {
```

Comment:
I think it would be nice to include a message here along the lines of "You have generated a single-use token for registering a new Bot Instance. This will be consumed after usage. Check out our delegated join methods (docs link) for a more secure and scaleable way to join new bot instances."

Reply:
I've added a new entry under "Please note:" with:

```
  - This is a single-token that will be consumed upon usage. For scalable
    alternatives, see our documentation on other supported join methods:
    https://goteleport.com/docs/enroll-resources/machine-id/deployment/
```

Reply:
Perfect!